// combat.proto
syntax = "proto3";

package combat;

import "common.proto";  // 导入状态枚举

// 实体类型（角色/怪物）
enum EntityType {
  CHARACTER = 0;
  MONSTER = 1;
}

// 攻击请求（角色或怪物发起攻击）
message AttackRequest {
  string entity_id = 1;       // 实体ID（角色/怪物）
  EntityType entity_type = 2; // 实体类型
  string target_id = 3;       // 目标ID（角色/怪物）
  float attack_angle = 4;     // 攻击方向角度
  string skill_id = 5;        // 使用的技能ID
  float cast_time = 6;        // 施法时间（秒）
}

// 受击事件（角色或怪物被攻击）
message HitEvent {
  string entity_id = 1;       // 被攻击实体ID
  EntityType entity_type = 2; // 被攻击实体类型
  string attacker_id = 3;     // 攻击者ID（角色/怪物）
  int32 damage = 4;           // 伤害值
  common.Status new_status = 5; // 新状态（如 DEAD、CASTING）
  common.MonsterState new_monster_state = 6; // 怪物的新状态（如 STUNNED、DEAD）
}